### ISSUE TITLE
Embedded table data

### ISSUE THREAD

**Issue by rheinheimer** [pywr/pywr#825]
I am trying to define a table within the "tables" section of the model json file by specifying the table data directly in the json file. Is this possible? If not, can it be?

I am looking at `dataframe_tools.py` file, noting [line 211](https://github.com/pywr/pywr/blob/1fd132209a6f871ff4883d82061cca6dca6e9acc/pywr/dataframe_tools.py#L211):
`df_data = data.pop('data', None)`, which is later loaded into a dataframe ([L257](https://github.com/pywr/pywr/blob/1fd132209a6f871ff4883d82061cca6dca6e9acc/pywr/dataframe_tools.py#L257)): `df = pandas.DataFrame.from_dict(df_data, **data)`. This implies as though if the table is formatted correctly in the json file, this should be possible. The alternative, of course (and arguably the better approach?) is to load normally from a CSV file, though the embedded approach is somewhat easier if bringing in a table from Hydra Platform.

Btw, the tables function is incredibly useful. I use this routinely for instream flow requirement lookup tables.

**Comment by jetuk** [comment]
Yes, this is possible. Something like this: https://github.com/pywr/pywr/blob/master/tests/models/simple_df_shared.json

Is this the question?

**Comment by rheinheimer** [comment]
In that example, the table data is defined indirectly, via a URL. What I mean is something like:
```
"tables": {
    "my_table": {
        "data": [[1,2,3],[4,5,6]],
        ...pandas kwargs
    }
}
```

Low priority for me. Saving to csv first is easy enough

**Comment by jetuk** [comment]
You can embed a pandas dataframe as per the test at the link below. It should work the same for a table as for a dataframe parameter.  Have a look at the code and how the `.to_json()` method works. However, pandas JSON format has some quirks here and there if you have complicated indices. 

https://github.com/pywr/pywr/blob/master/tests/test_parameters.py#L813

We should probably add a test for this particular case of an embedded JSON in the tables section though.
